Resources/config/controller.xml

```diff
@@ -24,6 +24,10 @@
             <argument>%hwi_oauth.connect%</argument>
             <argument>%hwi_oauth.grant_rule%</argument>
 
+            <call method="setContainer">
+                <argument type="service" id="service_container" />
```

Comment:
hmm this is incorrect. The `AbstractController` has a smaller service locator (its a `ServiceSubscriber`) and that is indeed currently not wired/configured correctly since https://github.com/hwi/HWIOAuthBundle/pull/1605 I think.

ping @niels-nijens did you try your change on a real app? It seems the controller is not autoconfigured correctly as a service subscriber and so the container will not be set properly by the symfony DI component?

Reply:
I might have forgotten about testing the change in a situation where you *are* using the `LoginController`... Sorry! To used to autowiring already...

Adding the `container.service_subscriber` tag should do the trick according to the documentation here:
https://symfony.com/doc/current/service_container/service_subscribers_locators.html#defining-a-service-subscriber (see the tip)

I will be able to test this tomorrow morning if you'd like.

Reply:
Thanks @niels-nijens . Would be great if you could test it and possibly open a new PR with a fix then if needed :blush:

Reply:
I've tested this in my application, but due to having autowiring and autoconfigure enabled in my project I wasn't able reproduce the problem and check if adding the tag would fix it. I'll try again in a fresh Symfony setup later this evening.

Reply:
Right, I've tested the `container.service_subscriber` and this creates a `ServiceLocator` with the requested services based on the `AbstractController::getSubscribedServices`, but... There seems no way to actually get the service locator injected without enabling autowiring on the `LoginController`. Something we don't want.

So, I guess adding the `setContainer` call with the service container is the way to go for now.

I think creating an issue on the Symfony repository might also be worthwhile, as it seems like something is missing for the 'controller in external bundle' use-case. Or could be that I overlooked something.

Reply:
> So, I guess adding the setContainer call with the service container is the way to go for now.

but this would not be correct.

1. service keys/ids can be aliased differently in the service locator than in the public container
2. not all services are publicly available in the service container

I just tried it myself and you are right. The controller must be autowired for it to work :confused: 

So the only solution I see for now is to add `autowire=true` for the `LoginController`. Its not the best solution but it won't break anything.